src/application/health_checks/health_checks.c

```diff
@@ -0,0 +1,196 @@
+#include "health_checks.h"
+#include "FreeRTOS.h"
+#include "application/can_handler/can_handler.h"
+#include "application/logger/log.h"
+#include "drivers/adc/adc.h"
+#include "drivers/timer/timer.h"
+#include "message_types.h"
+#include "printf.h"
+#include "task.h"
+
+#define TASK_DELAY_MS 1000
+#define ADC_VREF 3.3f
+#define R_SENSE 0.033f
+#define INA180A3_GAIN 100.0f
+#define MAX_CURRENT_mA 400
+#define MAX_WATCHDOG_TASKS 10
+#define E_WATCHDOG_TIMEOUT 0x81 // TODO PROPER BIT implementation
+#define CONV_ADC_COUNTS_TO_CURRENT_mA                                                              \
+    ((ADC_VREF * 1000.0f) / (ADC_MAX_COUNTS * INA180A3_GAIN * R_SENSE))
+
+// struct for watchdog
+typedef struct {
+    TaskHandle_t task_handle;
+    bool is_kicked;
+    float last_kick_timestamp;
+    uint32_t timeout_ticks;
+} watchdog_task_t;
+
+// watchdog initiailsations
+static watchdog_task_t watchdog_tasks[MAX_WATCHDOG_TASKS] = {0};
+static uint32_t num_watchdog_tasks = 0;
+
+w_status_t get_adc_current(uint32_t *adc_current_mA) {
+    w_status_t status = W_SUCCESS;
+    uint32_t adc_value;
+
+    status |= adc_get_value(PROCESSOR_BOARD_VOLTAGE, &adc_value, TASK_DELAY_MS);
+    if (status != W_SUCCESS) {
+        return status;
+    }
+
+    *adc_current_mA = (uint32_t)(adc_value * CONV_ADC_COUNTS_TO_CURRENT_mA);
+
+    return W_SUCCESS;
+}
+
+w_status_t check_current(void) {
+    uint32_t adc_current_mA;
+    w_status_t status = W_SUCCESS;
+
+    status |= get_adc_current(&adc_current_mA);
+    if (status == W_SUCCESS) {
```

Comment:
BARR says constant should be on the left of `==`:
```suggestion
    if (W_SUCCESS == status) {
```